x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/SqlClearCursorResponseListener.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.plugin;
+
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestResponseListener;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.sql.action.SqlClearCursorRequest;
+import org.elasticsearch.xpack.sql.action.SqlClearCursorResponse;
+
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.HEADER_NAME_TOOK_NANOS;
+
+public class SqlClearCursorResponseListener extends RestResponseListener<SqlClearCursorResponse> {
+    private final long startNanos = System.nanoTime();
+    private final RestRequest request;
+    private final XContentType responseType;
+
+    protected SqlClearCursorResponseListener(RestChannel channel, RestRequest request, SqlClearCursorRequest sqlRequest) {
+        super(channel);
+        this.request = request;
+        if (Boolean.TRUE.equals(sqlRequest.binaryCommunication())) {
+            responseType = XContentType.CBOR;
+        } else {
+            responseType = XContentType.JSON;
+        }
```

Comment:
I think we'll still want to keep the binary enforcement consistent with the query endpoint and return CBOR if Mode#isDriver() and the binary format is not explicitly disabled.